### DISCUSSION TITLE
How is NSM utilized in distributed setup?

### DISCUSSION THREAD

**Discussion by DinhDiesel** [Security-Onion-Solutions/securityonion#5987] [Unsupported Versions] [answered]
I noticed in Grafana my managersearch isnt pegged but at this rate memory and storage will be used up, I read in documentation that the heavy node is an extension of the manager+search...also noticed i have two wazuh managers. (i have all syslog & wazuh agents pointed to the managersearch) since i have two wazuh managers [manager+search + heavy] should i point them to the heavy node to really utilize the NSM (where i have much larger storage)?  not sure how that works on the back end, just curious.

**Comment by dougburks** [comment, marked as answer]
Heavy nodes are not really an extension of the manager in the same way that a normal search node is. Heavy nodes are typically not recommended and are only for specialized use cases. Some of your other discussions may have resulted from using a heavy node instead of a search node. You might want to review https://docs.securityonion.net/en/2.3/architecture.html and make sure you're using the appropriate architecture.

**Reply by DinhDiesel** [reply]
@dougburks  yes you are right in previous discussions my manager was only a manager which is now corrected, thank you very much! logs are logging as intended (sooooo grateful).  As far as the documentation I read "Similar to search nodes, heavy nodes extend the storage and processing capabilities of the manager node. However, heavy nodes also perform sensor duties and thus have lower performance overall." I feel that piece of information was misleading and my setup is backwards? I have a more resource hardware capable system as (heavy), less hardware capable (manager+search) in a distributed setup bc in the near future i want to add forward nodes to my grid.  Sounds like I should utilizie forward nodes instead of heavy and split my manager + search  node. I am just unsure as far as physical resources go for each node; Manager, Search, and Forward? (12GB, 4 cores, 200gb) for each node?

**Reply by dougburks** [reply]
> As far as the documentation I read "Similar to search nodes, heavy nodes extend the storage and processing capabilities of the manager node. However, heavy nodes also perform sensor duties and thus have lower performance overall." I feel that piece of information was misleading and my setup is backwards?

There is some nuance here. Heavy nodes are similar to search nodes in that you can search from the manager and get results from both search nodes and heavy nodes. However, they are different in that search nodes pull logs from the redis queue on the manager, while heavy nodes do not (they only store their own local logs). I've revised this section of the documentation to avoid confusion in the future.

> I have a more resource hardware capable system as (heavy), less hardware capable (manager+search) in a distributed setup bc in the near future i want to add forward nodes to my grid. Sounds like I should utilizie forward nodes instead of heavy and split my manager + search node.

The most powerful and scalable architecture is our standard distributed deployment with a manager, one or more search nodes, and one or more forward nodes:
https://docs.securityonion.net/en/2.3/architecture.html#distributed

> I am just unsure as far as physical resources go for each node; Manager, Search, and Forward? (12GB, 4 cores, 200gb) for each node?

Also see:
https://docs.securityonion.net/en/2.3/hardware.html#production-deployments
